src/core/lib/promise/cancel_callback.h

```diff
@@ -48,6 +51,8 @@ class Handler {
 
  private:
   Fn fn_;
+  RefCountedPtr<Arena> arena_ =
```

Comment:
Just for my own understanding, I'm guessing that the reason for this is that the `OnCancel()` callback might wind up being run after the arena context is removed, or maybe after the arena itself is freed?  Doesn't the `OnCancel()` callback get invoked as part of the `Party`, and if so, shouldn't the arena and its context always be present anyway?  Just trying to understand why we need to take special steps in this case.

Reply:
I'm not totally satisfied with this... but... cancellation happens at destruction time, and so we either get to whackamole every path that a promise might be destroyed on, or we capture enough to save context during that destruction... for now the sure fire thing seems pragmatic, but I hope to revisit this at some point and get a better story.

Reply:
Makes sense.  I suggest a TODO about this.